mobile/library/cc/engine_builder.cc

```diff
@@ -835,16 +835,20 @@ EngineSharedPtr EngineBuilder::build() {
 
   for (const auto& [name, store] : key_value_stores_) {
     // TODO(goaway): This leaks, but it's tied to the life of the engine.
-    auto* api = new envoy_kv_store();
-    *api = store->asEnvoyKeyValueStore();
-    register_platform_api(name.c_str(), api);
+    if (!Api::External::retrieveApi(name, true)) {
+      auto* api = new envoy_kv_store();
```

Comment:
so the asan issue was this new.  I was going to do the TODO and move ownership to the engine, but then realized we needed to atomically swap registered APIs if the engine went away and we're better off for multi-envoy support having one source of truth and not registering new ones.  So new version does that.